### PR TITLE
Highlight Shapes and Points layer data only when active

### LINKED ISSUE #7233: [UX] Shape & Points selection status remains visible when another layer is selected
When you draw a shape then it become selected when you finish (as of 0.5.0, https://github.com/napari/napari/pull/6640). This is great, because then you can change the Shape color, etc.
If you then switch the layer selection to an Image or Labels layer, the Shape continues to display the "selected" visual, even though it **cannot be interacted with**:
image

I feel like this is misleading and can result in stray painting or whatnot if the user tries to interact with the Shape without the Shapes layer selected.
Ideally, the selection visual would drop if a different layer is selected, but, importantly, return when the Shapes layer is re-selected.

Note in 0.4.19 shapes were not selected after drawing, so this potentially confusing effect didn't occur.
Edit: And in 0.4.19 if you manually select a shape and switch to a different layer it becomes de-selected. However, it is not be re-selected upon switching back to the Shapes layer.

Edit2: I reverted https://github.com/napari/napari/pull/6640 locally and Shapes are no longer selected on completion, but if you manually select a shape and change layers the selection visual still persists, so it must be related to a different PR.

Edit3: Selection of Points also persists visually when changing to a different layer type.

### PR DESCRIPTION
**Pull request by dalthviz**
# References and relevant issues

Closes #7233

# Description

Prevent showing data higlighted if the layers (Shapes or Points layers) are not the active layer:

* Shapes behavior:
shapes_highlight_active_layer

* Points behavior:

points_highlgiht

### CODE REVIEW

File: napari/components/viewer_model.py [hunk 1]
```diff
@@ -484,13 +484,16 @@ def _on_active_layer(self, event):
         if active_layer is None:
             for layer in self.layers:
                 layer.update_transform_box_visibility(False)
+                layer.update_highlight_visibility(False)
```
Comment (dalthviz) [inline napari/components/viewer_model.py:490]: Note: Maybe could it be worthy to unify these calls over a Layer method, something like a `_on_active_layer(is_active_layer: bool)` method? 🤔

Comment (DragaDoncila) [reply napari/components/viewer_model.py:490]: hmm it's not a bad idea, but I I feel like the layer shouldn't actually know whether it's active or not? That seems leaky to me but maybe others disagree.

**Review by DragaDoncila — COMMENTED** [review]
Confirmed this works, and changes look ok to me. Would be good to add a test for this, ideally.

**Comment by dalthviz** [comment]
> Would be good to add a test for this, ideally.

That could be nice indeed but not totally sure how to assert that something is not being rendered over the canvas. The only thing I can think of is maybe some sort of test where multiple layers are created and screenshots are taken to check for the color of the highlight while changing the active layer. Does something like that make sense or maybe there is a simplier alternative for a test? :thinking:

**Comment by DragaDoncila** [comment]
Hmm perhaps checking that the `highlight_visibility` value gets toggled correctly on active layer changed would be at least a partial test. Agreed it's harder to check if the highlight is actually rendered, I don't have good ideas there

File: napari/components/viewer_model.py (hunk 1 above)
Comment (jni) [reply napari/components/viewer_model.py:490]: Yeah, I think we are still struggling a bit figuring out where the "boundaries" between data, metadata, layers, and the UI are. And it's a high priority to come up with a consistent model of that.

But we do want *somewhere* to store this. It just might be a higher-level "LayerUI" object?

**Review by DragaDoncila — APPROVED** [review]
Thanks for adding that test @dalthviz, I think this is ready to go in

**Review by psobolewskiPhD — APPROVED** [review]
Thanks!
I think this approach makes sense and is straightforward. 
The highlight/layer/events interactions will need to be revisited based on some of the other issues (e.g. https://github.com/napari/napari/issues/7071), so for now lets go with this!

**Comment by jni** [comment]
> That could be nice indeed but not totally sure how to assert that something is not being rendered over the canvas.

typically I make a screenshot check with something that won't be too sensitive to rendering details. See for example [my test in 6654](https://github.com/napari/napari/pull/6654/files#diff-b34aee7c8386aa63aaa1809904227439f5f8ffb2469dc2939532564fdbaf78d6R573-R578):

```
    screenshot = viewer.screenshot(canvas_only=True, flash=False)
    # center should have a hole
    # use -1 because edge is exactly halfway down.
    center = (screenshot.shape[0] // 2 - 1, screenshot.shape[1] // 2)
    # ignore alpha channel
    np.testing.assert_array_equal(screenshot[center][:3], 0)
```

So in this case, I would check that the highlight color is or isn't present in the screenshot, based on the active layer.

**Comment by dalthviz** [comment]
> So in this case, I would check that the highlight color is or isn't present in the screenshot, based on the active layer.

Yep that's the initial idea that came to mind when thinking about creating a new test for this. Should I try to implement a new test doing screenshots then or the test changes I did following the flag check idea are enough?

**Comment by jni** [comment]
Personally I'd like to see such a test as it's more robust, but I think it can come in a follow-up PR. I'm gonna merge!

**Pull request merged**